src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infrastructure/ckan/utils/CkanAgentParser.java

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.discovery.datasets.infrastructure.ckan.utils;
+
+import io.github.genomicdatainfrastructure.discovery.model.Agent;
+import io.github.genomicdatainfrastructure.discovery.remote.ckan.model.CkanAgent;
+import lombok.experimental.UtilityClass;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Optional.ofNullable;
+
+@UtilityClass
+public class CkanAgentParser {
+
+    public List<Agent> agents(List<CkanAgent> agents) {
+        return ofNullable(agents)
```

Comment:
the use ofNullable here is a common anti pattern, just a null check would suffice